app/src/main/java/com/example/cleanarchitecture/mvvm/ListFromRetrofitViewModel.kt

```diff
@@ -0,0 +1,65 @@
+package com.example.cleanarchitecture.mvvm
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.domain.model.LceState
+import com.example.domain.model.Person
+import com.example.domain.usecase.GetPersonsFromApiUseCase
+import com.example.domain.usecase.InsertPersonToDatabaseUseCase
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class ListFromRetrofitViewModel(
+    private val getPersonsFromApiUseCase: GetPersonsFromApiUseCase,
+    private val insertPersonToDatabaseUseCase: InsertPersonToDatabaseUseCase
+) : ViewModel() {
+
+    private val _lceFlow = MutableStateFlow<LceState<List<Person>>>(LceState.Loading)
+    val lceFlow: Flow<LceState<List<Person>>> = _lceFlow.asStateFlow()
+
+    val _searchQueryFlow = MutableStateFlow("")
+    val searchQueryFlow = _searchQueryFlow.asStateFlow()
```

Comment:
в твоем случае наружу нужно выдавать не query flow, а то что ты написал ниже, тогда и метод по фильтрации можно будет сделать приватным
```suggestion
    val dataFlow = _searchQueryFlow
                .debounce(100)
                .mapLatest {
                    filterPersonList(it)
                }
```
хотя опять же не понятно почему у тебя есть метод который данные отдает и еще дополнительно Flow<Lce> который делает тоже самое